### PR TITLE
65 activate job seeker account by UUID

### PR DESCRIPTION
**Pull request by sharifyy**

### CODE REVIEW

File: src/main/kotlin/com/linchpino/core/controller/AccountController.kt [hunk 1]
```diff
@@ -75,4 +80,19 @@ class AccountController(private val accountService: AccountService) {
...
+
+    @Operation(summary = "Activate Job Seeker Account", description = "Activates a job seeker account")
+    @ApiResponse(
+        responseCode = "200", description = "Successfully activated job seeker account",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = AccountSummary::class))]
+    )
+    @PutMapping(
+        "/jobseeker/activation",
+        consumes = [MediaType.APPLICATION_JSON_VALUE],
+        produces = [MediaType.APPLICATION_JSON_VALUE]
```
Comment (m-tilab) [inline src/main/kotlin/com/linchpino/core/controller/AccountController.kt:92]: Can you tell me why we need to mention the JSON value as content type? I think its a default value.

**Review by m-tilab — COMMENTED** [review]
Please merge development also

File: src/main/kotlin/com/linchpino/core/controller/AccountController.kt (hunk 1 above)
Comment (sharifyy) [reply src/main/kotlin/com/linchpino/core/controller/AccountController.kt:92]: Just for the sake of clarity of the documentation. If you think MediaType is unnecessary I will remove it.

Comment (sharifyy) [reply src/main/kotlin/com/linchpino/core/controller/AccountController.kt:92]: The branch is up to date now.

File: src/main/kotlin/com/linchpino/core/service/AccountService.kt [hunk 2]
```diff
@@ -35,6 +40,21 @@ class AccountService(
...
+
+    fun activeJobSeekerAccount(request: ActivateJobSeekerAccountRequest): AccountSummary {
+        val account = repository.findByExternalId(request.externalId, AccountTypeEnum.JOB_SEEKER)
+            ?: throw RuntimeException("account not found")
+        if (account.status == AccountStatusEnum.ACTIVATED)
+            throw RuntimeException("account is already activated")
+        val updatedAccount = account.apply {
+            firstName = request.firstName
+            lastName = request.lastName
+            password = request.password
```
Comment (m-tilab) [inline src/main/kotlin/com/linchpino/core/service/AccountService.kt]: The password is not encrypted. I have defined another issue to refactor the add account to point all places to a generic point. I mean only one point should call accountRepository.save

Comment (sharifyy) [reply src/main/kotlin/com/linchpino/core/service/AccountService.kt]: I updated the pr with encoded password for now.

File: src/test/kotlin/com/linchpino/core/service/AccountServiceTest.kt [hunk 3]
```diff
@@ -109,4 +112,73 @@ class AccountServiceTest {
...
+        }
+
+        `when`(repository.findByExternalId(request.externalId, AccountTypeEnum.JOB_SEEKER)).thenReturn(account)
+        val result = accountService.activeJobSeekerAccount(request)
+
+        verify(repository, times(1)).save(account)
+
+        assertThat(result.firstName).isEqualTo(result.firstName)
+        assertThat(result.lastName).isEqualTo(result.lastName)
+        assertThat(result.status).isEqualTo(AccountStatusEnum.ACTIVATED)
```
Comment (m-tilab) [inline src/test/kotlin/com/linchpino/core/service/AccountServiceTest.kt:141]: Please verify the password that is encrypted

**Review by m-tilab — APPROVED** [review]

**Pull request merged**